community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/tests/test_util.py

```diff
@@ -308,3 +311,36 @@ def test_nodeset_reservation_ok(nodeset, policies, expected):
 def test_parse_job_info(job_info, expected_job):
     lkp = util.Lookup(TstCfg())
     assert lkp._parse_job_info(job_info) == expected_job
+
+
+
+@pytest.mark.parametrize(
+    "node,state,want",
+    [
+        ("c-n-2", NodeState("DOWN", {}), NodeState("DOWN", {})), # happy scenario
+        ("c-d-vodoo", None, None), # dynamic nodeset
+        ("c-x-44", None, None), # unknown(removed) nodeset
+        ("c-n-7", None, None), # Out of bounds: c-n-[0-4] - downsized nodeset
+        ("c-t-7", None, None), # Out of bounds: c-t-[0-4] - downsized nodeset TPU
```

Comment:
Why not test c-n-5 on one of these to hit the limit case, protect against off by one error.